src/exec/command.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2017 fd developers
+// Licensed under the Apache License, Version 2.0
+// <LICENSE-APACHE or http://www.apache.org/licenses/LICENSE-2.0>
+// or the MIT license <LICENSE-MIT or http://opensource.org/licenses/MIT>,
+// at your option. All files in the project carrying such
+// notice may not be copied, modified, or distributed except
+// according to those terms.
+
+use std::process::Command;
+use std::sync::{Arc, Mutex};
+use std::io;
+
+/// Executes a command.
+#[cfg(not(unix))]
+pub fn execute_command(mut cmd: Command, out_perm: Arc<Mutex<()>>) {
+    use std::process::Stdio;
+    use std::io::Write;
+
+    // Spawn the supplied command.
+    let output = cmd.stdout(Stdio::piped()).stderr(Stdio::piped()).output();
+
+    // Then wait for the command to exit, if it was spawned.
+    match output {
+        Ok(output) => {
+            // While this lock is active, this thread will be the only thread allowed
+            // to write it's outputs.
+            let _lock = out_perm.lock().unwrap();
+
+            let stdout = io::stdout();
+            let stderr = io::stderr();
+
+            let _ = stdout.lock().write_all(&output.stdout);
+            let _ = stderr.lock().write_all(&output.stderr);
+        }
+        Err(why) => eprintln!("fd: exec error: {}", why),
```

Comment:
It would be great if we could improve this error message here. I'm not sure what all the possible cases are that could trigger this error, but this message will be shown to the user if the given command is not available:
```
> fd -x foo
fd: exec error: No such file or directory (os error 2)
```
I encountered this by accident and was definitely confused until I realized that the command I specified did not exist.

Reply:
This is probably out of scope for this issue. While this code is all marked as new, it's actually what is left from the renamed and modified `ticket.rs`.

Reply:
You'll have to construct the error message yourself, by checking the error kind of the error.

```rust
if why.kind() == io::ErrorKind::NotFound {
    eprintln!("fd: exec error: command not found");
} else {
    eprintln!("fd: exec error: {}", why);
}
```

Reply:
Thanks @mmstick. @reima Oh, I missed that. I'll do it after merging this PR.